File: ProductAPI/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductAPI.Models
{
    public class Product
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string ProductId { get; set; }
        public string ProductName { get; set; }
        public string ProductType { get; set; }
        public string ProductByGender { get; set; }
        public string ProductDescription { get; set; }
        public string ProductBrand { get; set; }
        public string ProductSize { get; set; }
        public string ProductColor { get; set; }
        public float ProductPrice { get; set; }
        public int ProductStock { get; set; }
    }
}

File: jeanstationfrontend/src/app/models/Product.ts
import ProductImage from "./ProductImage";

export default class Product{
    productId: string;
    productName: string;
    productType: string;
    productByGender: string;
    productDescription: string;
    productBrand: string;
    productSize: string;
    productColor: string;
    productImageId: string;
    productPrice: number;
    ProductImages: Array<ProductImage>;
    productStock: number;

    constructor(){
        this.productId = "";
        this.productName = "";
        this.productType = "";
        this.productByGender = "";
        this.productDescription = "";
        this.productBrand = "";
        this.productSize = "";
        this.productColor= "";
        this.productImageId = "";
        this.productPrice = 0;
        this.ProductImages = [];
        this.productStock = 0;
    }
}